src/expression.js

```diff
@@ -706,7 +706,7 @@ pp.parseFunctionBody = function(node, isArrowFunction) {
     // flag (restore them to their old value afterwards).
     let oldLabels = this.labels
     let oldVarDeclaredNames = this.varDeclaredNames
-    this.varDeclaredNames = new Set()
+    this.varDeclaredNames = {}
```

Comment:
You can use `Object.create(null)` so that the `hasOwnProperty()` checks won't be necessary